src/routes/wishlistRoutes.js

```diff
@@ -0,0 +1,79 @@
+const express = require('express');
+const Wishlist = require('./Wishlist');
+const Album = require('./Album');
+const router = express.Router();
+const mongoose = require('mongoose');
+
+router.patch('/wishlist/:wishlist_id/add_album/:album_id', async (req, res) => {
+  const { wishlist_id, album_id } = req.params;
+
+  try {
+    if (
+      !mongoose.Types.ObjectId.isValid(wishlist_id) ||
+      !mongoose.Types.ObjectId.isValid(album_id)
+    ) {
+      return res.status(400).send('Invalid wishlist_id or album_id');
+    }
+
+    // Find the Wishlist and Album documents
+    const wishlist = await Wishlist.findById(wishlist_id);
+    const album = await Album.findById(album_id);
+
+    // Check if wishlist and album exist
+    if (!wishlist || !album) {
+      return res.status(404).send('Wishlist or album not found');
+    }
+
+    // Check if album is already in wishlist
+    if (wishlist.albums.includes(album._id)) {
+      return res.status(200).json(wishlist);
+    }
+
+    // Add album to wishlist and save
+    wishlist.albums.push(album._id);
+    await wishlist.save();
+
+    return res.status(200).json(wishlist);
+  } catch (error) {
+    return res.status(500).send('Server error');
+  }
```

Comment:
```suggestion
    if (
      !mongoose.Types.ObjectId.isValid(wishlist_id) ||
      !mongoose.Types.ObjectId.isValid(album_id)
    ) {
      return res.status(400).send('Invalid wishlist_id or album_id');
    }

    // Find the Wishlist and Album documents
    const wishlist = await Wishlist.findById(wishlist_id);
    const album = await Album.findById(album_id);

    // Check if wishlist and album exist
    if (!wishlist || !album) {
      return res.status(404).send('Wishlist or album not found');
    }

    // Check if album is already in wishlist
    if (wishlist.albums.includes(album._id)) {
      return res.status(200).json(wishlist);
    }

    // Add album to wishlist and save
    wishlist.albums.push(album._id);
    await wishlist.save();

    return res.status(200).json(wishlist);
```

We shouldn't need to wrap our controller code in try/catch because we're using the `express-async-errors` library. Any erorr sthat happen and get thrown in here, will be handled by the errorHandler middleware file.

Reply:
Yes , you are right, thank you , commit 489aa8e60f2ee602eafdc253ead5c9ed92ad8495